### ISSUE TITLE
Sensible wrapping, where appropriate

### ISSUE THREAD

**Issue by JakeWharton** [square/javapoet#105]
There are parts of code to which we can apply sensible wrapping. Right column break to be configured on `JavaWriter`.

Candidates include:
- Class declaration (extends, interfaces)
- Field initializer (if length exceeds wrap&indent initializer)
- Method declaration (params, exceptions)

**Comment by roykachouh** [comment]
This was exactly what I was looking for. Any updates on when to expect this. I'm using JavaPoet (which i love!) to automatically generate server stubs and clients for my project based on specifications. 

There are times when things look a little messy, think builder pattern with a constructor with 10 arguments. Ideally, I was hoping to find something that can take checkstyle and automatically reformat based on our coding conventions, but it doesn't seem there is an appropriate solution.

Is this enhancement still on your milestone list?

**Comment by roykachouh** [comment]
I can also try to take a shot at this with a pr.

**Comment by swankjesse** [comment]
Really difficult to do well. Some teams have recommended coupling JavaPoet with another code formatter.

**Comment by roykachouh** [comment]
I agree. That would be ideal as there can potentially be so many different code style preferences. Do you happen to know of any promising auto format tools out there? Googling has come up short for me so far.

**Comment by tbroyer** [comment]
@roykachouh Google has https://github.com/google/google-java-format specific to their style.

**Comment by sormuras** [comment]
They released version [1.0 Alpha](https://github.com/google/google-java-format/releases) with an all-inclusive jar. Just include it in your project and add a last line like:
```java
String formattedSource = new Formatter().formatSource(javapoetJavaFile.toString());
```
